BUG: Getting the server to run on windows.

prabhuramachandran opened the pull request:
Windows does not support the AF_UNIX family so using AF_INET instead.  There are some aspects of this pull request that are not clean, notably the (ab)use of eval in server.py.  Basically, if the address passed starts with a '(' I treat it as an AF_INET family and eval it to get the right address.   The other issue, is that unlike the AF_UNIX case I do not poll the socket to check if the server has started or not -- this was because I wasn't able to quickly figure out a clean way to do that.

In any case this pull-request seems to fix issues with running supplement on win32.

prabhuramachandran committed:
BUG: Getting the server to run on windows.

Windows does not support the AF_UNIX family so using AF_INET instead.

baverman commented:
I'm sorry for long answer.

I reviewed your pull request and I think I found an easier way to create duplex channel under win through named pipes. Also I removed platform-depended method to wait server listener. Please check it out.

prabhuramachandran commented:
Hi,

On Wed, Oct 12, 2011 at 5:18 PM, Anton Bobrov <
reply@reply.github.com>wrote:

> I'm sorry for long answer.

No problem.

> I reviewed your pull request and I think I found an easier way to create
> duplex channel under win through named pipes. Also I removed
> platform-depended method to wait server listener. Please check it out.

Yes, this is much cleaner and seems to work fine.  Thanks!  I have locally
fixed another little bug and will submit a pull request.

cheers,
Prabhu
